org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/RenameRequest.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4xml.services;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.dom.DOMElement;
+import org.eclipse.lsp4xml.services.extensions.IRenameRequest;
+
+class RenameRequest implements IRenameRequest {
+
+	private final DOMDocument document;
+	private final DOMElement element;
+	private final Position position;
+	private final String newText;
+	
+	public RenameRequest(DOMDocument document, DOMElement element, Position position, String newText) {
+		this.document = document;
+		this.element = element;
+		this.position = position;
+		this.newText = newText;
+	}
+
+	@Override
+	public DOMDocument getDocument() {
+		return document;
+	}
+
+	@Override
+	public DOMElement getElement() {
```

Comment:
remove this method which exist sin AbstractPositionRequest. It is called getNode which returns the element or the attribute at the given offset